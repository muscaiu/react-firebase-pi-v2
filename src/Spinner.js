import React from 'react';
import styled, { keyframes } from 'styled-components';

import logo from './logo.svg';

const Wrapper = styled.div`
`;

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Logo = styled.img`
  animation: ${props => props.isActive ? `${rotate360} infinite 20s linear` : 'null'};
  height: 150px;
`;

const Spinner = ({ isActive }) => (
  <Wrapper>
    <Logo isActive={isActive} src={logo} />
  </Wrapper>
)

export default Spinner;
