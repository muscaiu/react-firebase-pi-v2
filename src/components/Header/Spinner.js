import React from 'react';

const Spinner = () => (
  <div className="preloader">
    <div></div>
  </div>
)

export default Spinner;
