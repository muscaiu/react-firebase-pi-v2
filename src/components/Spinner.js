import React from 'react';

const Spinner = ({ isActive }) => (
  <div className="preloader">
    <div></div>
  </div>
)

export default Spinner;
