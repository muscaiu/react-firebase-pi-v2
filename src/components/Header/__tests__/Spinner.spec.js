import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from '../Header';

describe('Header', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Spinner />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
