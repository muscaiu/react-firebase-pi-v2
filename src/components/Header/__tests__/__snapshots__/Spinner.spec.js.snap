// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match snapshot 1`] = `
<div
  className="preloader"
>
  <div />
</div>
`;
